### PR TITLE
Add Missing Type Annotations

### LINKED ISSUE #10: Missing Type Annotations
There are missing type annotations for some functions in this library.

The `typing` module does not exist on CircuitPython devices so the import needs to be wrapped in try/except to catch the error for missing import. There is an example of how that is done here:

```python
try:
    from typing import List, Tuple
except ImportError:
    pass
```

Once imported the typing annotations for the argument type(s), and return type(s) can be added to the function signature. Here is an example of a function that has had this done already:

```python
def wrap_text_to_pixels(
    string: str, max_width: int, font=None, indent0: str = "", indent1: str = ""
) -> List[str]:
```
If you are new to Git or Github we have a guide about contributing to our projects here: https://learn.adafruit.com/contribute-to-circuitpython-with-git-and-github

There is also a guide that covers our CI utilities and how to run them locally to ensure they will pass in Github Actions here: https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library/check-your-code In particular the pages: `Sharing docs on ReadTheDocs` and `Check your code with pre-commit` contain the tools to install and commands to run locally to run the checks.

If you are attempting to resolve this issue and need help, you can post a comment on this issue and tag both @foamyguy and @kattni or reach out to us on Discord: https://adafru.it/discord in the `#circuitpython-dev` channel.

The following locations are reported by mypy to be missing type annotations:

- [ ] adafruit_tfmini.py:48
- [ ] adafruit_tfmini.py:100
- [ ] adafruit_tfmini.py:105

### PR DESCRIPTION
**Pull request by tcfranks**
resolves #10 
submitted for review / comment

### CODE REVIEW

File: adafruit_tfmini.py [hunk 1]
```diff
@@ -46,7 +52,7 @@ class TFmini:
     :param timeout: how long we'll wait for valid data or response, in seconds. Default is 1
     """
 
-    def __init__(self, uart, *, timeout=1):
+    def __init__(self, uart: UART, *, timeout: int = 1) -> None:
```
Comment (tekktrik) [inline adafruit_tfmini.py]: Since `timeout` gets used with `time.monotonic()` comparisons, we can type it as `float`.

File: adafruit_tfmini.py [hunk 2]
```diff
@@ -55,7 +61,7 @@ def __init__(self, uart, *, timeout=1):
         self._mode = None
 
     @property
-    def distance(self):
+    def distance(self) -> None:
```
Comment (tekktrik) [inline adafruit_tfmini.py]: Looks like it returns `dist`, which is an `int`

File: adafruit_tfmini.py [hunk 3]
```diff
@@ -87,24 +93,24 @@ def distance(self):
...
     @property
-    def strength(self):
+    def strength(self) -> int:
         """The signal validity, higher value means better measurement"""
         _ = self.distance  # trigger distance measurement
         return self._strength
 
     @property
-    def mode(self):
+    def mode(self) -> int:
```
Comment (tekktrik) [inline adafruit_tfmini.py]: Since it's only a couple of things here, let's use `Literal`.

File: adafruit_tfmini.py [hunk 4]
```diff
@@ -87,24 +93,24 @@ def distance(self):
...
         """The measurement mode can be MODE_SHORT (2) or MODE_LONG (7)"""
         _ = self.distance  # trigger distance measurement
         return self._mode
 
     @mode.setter
-    def mode(self, newmode):
+    def mode(self, newmode: int) -> None:
```
Comment (tekktrik) [inline adafruit_tfmini.py]: Same thing here, let's use `Literal`

File: adafruit_tfmini.py [hunk 5]
```diff
@@ -87,24 +93,24 @@ def distance(self):
...
         if not newmode in (MODE_LONG, MODE_SHORT):
             raise ValueError("Invalid mode")
         self._set_config(_CONFIGPARAM + bytes([0, 0, newmode, 0x11]))
 
-    def _set_config(self, command):
+    def _set_config(self, command: int) -> None:
```
Comment (tekktrik) [inline adafruit_tfmini.py]: `command` can be `ReadableBuffer` from `circuitpython_typing` from `adafruit-circuitpython-typing`.

**Review by tekktrik — CHANGES_REQUESTED** [review]

File: adafruit_tfmini.py (hunk 2 above)
Comment (tcfranks) [reply adafruit_tfmini.py]: looks like it could return nothing and just raise an error as well.....

**Comment by tcfranks** [comment]
resubmitted with changes

File: adafruit_tfmini.py (hunk 2 above)
Comment (tekktrik) [reply adafruit_tfmini.py]: Exceptions don't get included in the type annotation if there is a valid return (even `None`).  If the only thing a function does with 100% certainty is raise an exception (so not even return `None`), we can type annotate it as `NoReturn`.  Doesn't apply here, but seemed relevant to your point.

Comment (tcfranks) [reply adafruit_tfmini.py]: thanks for the clarification, Alec

File: adafruit_tfmini.py [hunk 6]
```diff
@@ -99,18 +101,18 @@ def strength(self) -> int:
...
     @property
-    def mode(self) -> int:
+    def mode(self) -> Literal[2, 7]:
         """The measurement mode can be MODE_SHORT (2) or MODE_LONG (7)"""
         _ = self.distance  # trigger distance measurement
         return self._mode
 
     @mode.setter
-    def mode(self, newmode: int) -> None:
+    def mode(self, newmode: Literal[MODE_LONG, MODE_SHORT]) -> None:
```
Comment (tekktrik) [inline adafruit_tfmini.py]: I don't believe that using variables is valid use of `Literal`  Let's make it `Literal[2, 7]` like the getter.

**Review by tekktrik — CHANGES_REQUESTED** [review]
One small change:

**Comment by tcfranks** [comment]
removed constant names as indicated

**Review by tekktrik — APPROVED** [review]
LGTM, thanks!

**Pull request merged**